primefaces/src/main/resources/META-INF/resources/primefaces/core/core.env.js

```diff
@@ -70,6 +99,26 @@ if (!PrimeFaces.env) {
         isTouchable: function(cfg) {
             var widgetTouchable = (cfg == undefined) || (cfg.touchable != undefined ? cfg.touchable : true);
             return PrimeFaces.env.touch && widgetTouchable;
+        },
+
+        /**
+         * Gets the user's preferred color scheme set in their operating system.
+         * 
+         * @return {string} either 'dark' or 'light'
+         */
+        getOSPreferredColorScheme: function() {
+            return PrimeFaces.env.preferredColorSchemeLight ? 'light' : 'dark';
+        },
+
+       /**
+         * Based on the current PrimeFaces theme determine if light or dark contrast is being applied.
+         * 
+         * @return {string} either 'dark' or 'light'
+         */
+        getThemeContrast: function() {
+            var theme = PrimeFaces.env.getTheme();
+            var darkRegex = /(arya|vela|\S+-dim|\S+-dark)/gm;
```

Comment:
Shouldn't this be
```suggestion
            var darkRegex = /(^(arya|vela|.+-(dim|dark))$)/gm;
```

or simply
```
return (/(^(arya|vela|.+-(dim|dark))$)/gm).test(PrimeFaces.env.getTheme()) ? 'dark' : 'light';
```

Reply:
Updated the regex.  I like leaving the separate lines it makes it easier to step through with the debugger than on one long line.

Reply:
Okay. In Chrome you can have multiple breakpoint on single line I think. I agree this is more readable.